### PR TITLE
Feature/register service connect to server

### PR DESCRIPTION
**Pull request by tee0paw**

### CODE REVIEW

File: roadmap.org [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
-** TODO Implement Groups Overview Screen
-*** TODO Show all groups from user
-    Groups should show small summary how much the user owes or paid
-    for the group.
-*** TODO Allow to edit groups
-*** TODO Allow to delete groups
-
-** TODO Implement Friends Overview Screen
-*** TODO Show all friends from user
-    Friends should show small summary how much user owes or paid.
```
Comment (FlexW) [inline roadmap.org:18]: Unrelated change :)

File: src/app/_helpers/fake-backend.ts [hunk 2]
```diff
@@ -301,3 +301,4 @@ export const fakeBackendProvider = {
   useClass: FakeBackendInterceptor,
   multi: true,
 };
+*/
```
Comment (FlexW) [inline src/app/_helpers/fake-backend.ts]: Please delete fake-backend.ts :)

File: src/app/_helpers/index.ts [hunk 3]
```diff
@@ -1,3 +1,3 @@
-export * from './fake-backend';
+//export * from './fake-backend';
```
Comment (FlexW) [inline src/app/_helpers/index.ts]: Dito

File: src/app/_services/bills.service.spec.ts [hunk 4]
```diff
@@ -44,13 +44,13 @@ describe('BillsService', () => {
       },
     ];
 
-    service.getBillsWithUsersByUserId(user_id).subscribe((result) => {
+    service.getBillsWithUsersByUserId().subscribe((result) => {
```
Comment (FlexW) [inline src/app/_services/bills.service.spec.ts:47]: unrelated

File: src/app/_services/bills.service.ts [hunk 5]
```diff
@@ -11,22 +11,22 @@ interface BillsWithUsersRequestResult {
   bills: BillWithUsers[];
 }
 
+export interface BillsRequestResult {
+  message: string;
+  group: { id: number };
+}
+
```
Comment (FlexW) [inline src/app/_services/bills.service.ts]: unrelated

File: src/app/_services/bills.service.ts [hunk 6]
```diff
@@ -11,22 +11,22 @@ interface BillsWithUsersRequestResult {
...
 @Injectable({
   providedIn: 'root',
 })
 export class BillsService {
   constructor(private http: HttpClient) {}
 
-  getBillsWithUsersByUserId(
-    userId: number
-  ): Observable<BillWithUsers[] | RequestError> {
+  getBillsWithUsersByUserId(): Observable<BillWithUsers[] | RequestError> {
```
Comment (FlexW) [inline src/app/_services/bills.service.ts:20]: unrelated

File: src/app/_services/bills.service.ts [hunk 7]
```diff
@@ -11,22 +11,22 @@ interface BillsWithUsersRequestResult {
...
     return this.http
-      .get<BillsWithUsersRequestResult>(
-        `${environment.apiUrl}/bills/${userId}`,
-        {
-          headers: getHeaders(),
-        }
-      )
+      .get<BillsWithUsersRequestResult>(`${environment.apiUrl}/bills`, {
+        headers: getHeaders(),
+      })
```
Comment (FlexW) [inline src/app/_services/bills.service.ts:24]: unrelated

File: src/app/_services/register.service.ts [hunk 8]
```diff
@@ -1,7 +1,7 @@
 import { HttpClient, HttpHeaders } from '@angular/common/http';
 import { Injectable } from '@angular/core';
-import { Observable } from 'rxjs';
-import { map } from 'rxjs/operators';
+import { Observable, throwError } from 'rxjs';
+import { map, catchError, retry } from 'rxjs/operators';
```
Comment (FlexW) [inline src/app/_services/register.service.ts]: unrelated

File: src/app/app.module.ts [hunk 9]
```diff
@@ -28,7 +28,7 @@ import { OverviewComponent } from './overview/overview.component';
 import { MatExpansionModule } from '@angular/material/expansion';
 
 // used to create fake backend
-import { fakeBackendProvider } from './_helpers';
+//import { fakeBackendProvider } from './_helpers';
```
Comment (FlexW) [inline src/app/app.module.ts]: please remove comment

File: src/app/app.module.ts [hunk 10]
```diff
@@ -70,7 +70,7 @@ import { GroupListItemComponent } from './group-list-item/group-list-item.compon
   ],
   providers: [
     { provide: HTTP_INTERCEPTORS, useClass: JwtInterceptor, multi: true },
-    fakeBackendProvider,
+    //fakeBackendProvider,
```
Comment (FlexW) [inline src/app/app.module.ts]: remove

File: src/app/bills/bills.component.ts [hunk 11]
```diff
@@ -34,12 +34,10 @@ export class BillsComponent implements OnInit {
...
-      .subscribe((result) => {
-        if (!(result instanceof RequestError)) {
-          this.bills = result;
-        }
-      });
+    this.billsService.getBillsWithUsersByUserId().subscribe((result) => {
+      if (!(result instanceof RequestError)) {
+        this.bills = result;
+      }
+    });
```
Comment (FlexW) [inline src/app/bills/bills.component.ts]: unrelated

File: src/app/group-list-item/group-list-item.component.ts [hunk 12]
```diff
@@ -50,10 +50,7 @@ export class GroupListItemComponent implements OnInit {
       const lastName = member.lastName;
       const label = `${firstName} ${lastName} is a member.`;
 
-      let style = { color: 'red' };
-      if (this._currentUserId === member.id) {
-        style = { color: 'green' };
-      }
+      let style = { color: 'black' };
```
Comment (FlexW) [inline src/app/group-list-item/group-list-item.component.ts:53]: unrelated

**Review by FlexW — CHANGES_REQUESTED** [review]

File: src/app/_services/bills.service.spec.ts (hunk 4 above)
Comment (tee0paw) [reply src/app/_services/bills.service.spec.ts:47]: cannot delete this, this throws error

**Pull request closed without merging**